File: source/code/csharp/authentication/create_directory_google.cs
using Stormpath.SDK.Provider;

var googleDirectory = client.Instantiate<IDirectory>();
googleDirectory.SetName("My Google Directory");

await client.CreateDirectoryAsync(googleDirectory,
    options => options.ForProvider(
        client.Providers()
            .Google()
            .Builder()
            .SetClientId("YOUR_GOOGLE_CLIENT_ID")
            .SetClientSecret("YOUR_GOOGLE_CLIENT_SECRET")
            .SetRedirectUri("YOUR_GOOGLE_REDIRECT_URI")
            .Build()
        ));

File: source/code/java/authentication/create_directory_google.java
Directory directory = client.instantiate(Directory.class);
directory.setName("My Google Directory");
directory.setDescription("A Google directory");

CreateDirectoryRequest request = Directories.newCreateRequestFor(directory)
    .forProvider(Providers.GOOGLE.builder()
        .setClientId("YOUR_GOOGLE_CLIENT_ID")
        .setClientSecret("YOUR_GOOGLE_CLIENT_SECRET")
        .setRedirectUri("YOUR_GOOGLE_REDIRECT_URI")
        .build()
    ).build();

directory = client.createDirectory(request);
